### DISCUSSION TITLE
Watch Replay

### DISCUSSION THREAD

**Discussion by lukew3** [monkeytypegame/monkeytype#1082] [Ideas] [answered]
There should be a way that you can watch a replay of a test after it has been completed. Stats like current speed and error count could be shown and updated as playback progresses. 

Somebody might use this if they want to analyze a test or re-watch a particularly fast test.

**Comment by Miodec** [comment, marked as answer]
This was already rejected due to storage reasons. Keeping information for all tests taken for all users would get really big really quickly.

**Reply by lukew3** [reply]
I understand. However, this could still be implemented in the frontend similar to how the toggle words history works.

**Reply by unrealapex** [reply]
@Miodec Would it be possible to store the replay temporarily, having the user view the replay after a test, then the replay gets deleted once the user goes on to the next test ?

**Reply by Miodec** [reply]
Right, so just have it locally for one test?

**Reply by lukew3** [reply]
@Miodec Yeah, I could try to build it right now if you approve.

**Reply by Miodec** [reply]
Yeah, sure go for it. I'm assuming you're going to integrate it into the words history? Add a little play pause repeat buttons next to the copy text icon?

![image](https://user-images.githubusercontent.com/13181393/110952978-c0282700-833e-11eb-91cb-51955e2cf581.png)

**Reply by lukew3** [reply]
I was thinking a seperate button on that toolbar. When clicked, a playback overlay would go where the test results were. You could exit out of this to go back to the results when you are done.

**Reply by Godje** [reply]
@lukew3 I'd say Miodec's suggestion for the UI thing is better.
Having that would make sense of the error highlights on the bottom, and wouldn't bother with redrawing the graph (since its a canvas)

**Reply by unrealapex** [reply]
> @lukew3 I'd say Miodec's suggestion for the UI thing is better.
> Having that would make sense of the error highlights on the bottom, and wouldn't bother with redrawing the graph (since its a canvas)

Agreed

**Reply by Miodec** [reply]
Actually yeah, maybe a new button on the toolbar, to the right of screenshot, and keep the replay in a popup.

**Reply by lukew3** [reply]
Made a pull request for a version similar to the first proposed example. #1089

**Reply by lukew3** [reply]
New pr #1125
